### DISCUSSION TITLE
How to extract the entire current path in a handler function instead of extracting parameters from the path.

### DISCUSSION THREAD

**Discussion by graphsay** [tokio-rs/axum#2652] [Q&A] [answered]
### Summary

How to extract the entire current path in a handler function instead of extracting parameters from the path. 
Because I have multiple routes point to a same handler function, it is necessary to distinguish the current URL path in the handler.
thanks for help.

### axum version

0.7.4

**Comment by jplatte** [comment, marked as answer]
You can add a `http::Uri` parameter to your handler and get the path from that.

You can also use different handler functions that just call the same inner function immediately that does the actual work, while passing an extra parameter for route-specific information.

**Reply by graphsay** [reply]
Yes, this is exactly what I want and has solved my problem. Thank you very much.
